src/main/java/domain/piece/General.java

```diff
@@ -0,0 +1,32 @@
+package domain.piece;
+
+import domain.board.Position;
+
+import java.util.List;
+
+public class General extends Piece {
+
+    public General(PieceColor color) {
+        super(color);
+    }
+
+    @Override
+    public boolean isValidDestination(Position source, Position destination) {
+        return false;
+    }
+
+    @Override
+    public List<Position> findAllRoute(Position source, Position destination) {
+        return List.of();
+    }
+
+    @Override
+    public boolean canMove(Piece destinationPiece, List<Piece> piecesInRoute) {
+        int pieceCountInRoute = this.countPieceInRoute(piecesInRoute);
+
+        if (this.isOtherTeam(destinationPiece) && pieceCountInRoute == 0) {
+            return true;
+        }
+        return false;
```

Comment:
```suggestion
        return this.isOtherTeam(destinationPiece) && pieceCountInRoute == 0;
```
요렇게 단순화시킬수 있을 것 같아요.